### PR TITLE
[Cider/Interpreter] Source position, misc bug fixing, bounds checking

### PR DESCRIPTION
**Pull request by EclecticGriffin**
This adds the rudimentary source position tags, fixes a number of state machine bugs, also includes bounds checking warning per #900.

### CODE REVIEW

File: fud/fud/config.py [hunk 1]
```diff
@@ -32,8 +32,8 @@
             "flags": None,
             "data": None,
             "round_float_to_fixed": True,
-            "debugger": {"flags": None},
         },
+        "debugger": {"flags": None},
```
[the remark this answers is not in the document]
Comment (EclecticGriffin) [reply fud/fud/config.py:36]: Moved it since it was confusing

File: Cargo.toml [hunk 2]
```diff
@@ -31,10 +31,10 @@ members = ["calyx", "interp", "web/rust"]
 exclude = ["site"]
 
 [profile.release]
-incremental = true
+lto = true
```
[the remark this answers is not in the document]
Comment (EclecticGriffin) [reply Cargo.toml:34]: LTO does increase release build times. I'm not sure I understand? It is under the release profile, no?

[the remark this answers is not in the document]
Comment (rachitnigam) [reply Cargo.toml:34]: Ignore me, I am blind

File: interp/Cargo.toml [hunk 3]
```diff
@@ -11,6 +11,7 @@ doctest = false # Don't run doc tests
 
 [features]
 change-based-sim = []
+default = ["change-based-sim"]
```
[the remark this answers is not in the document]
Comment (EclecticGriffin) [reply interp/Cargo.toml:14]: Good point, I'll touch up the Interpreter's README since it's rather old

**Pull request merged**